internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go

```diff
@@ -495,7 +495,7 @@ func expandTKGSVolumes(data interface{}) (volume *nodepoolmodel.VmwareTanzuManag
 	volume = &nodepoolmodel.VmwareTanzuManageV1alpha1CommonClusterTKGServiceVsphereVolume{}
 
 	if v, ok := lookUpVolumes[capacityKey]; ok {
-		volume.Capacity, _ = v.(float32)
+		helper.ReadPrimitiveTypes(v, &volume.Capacity)
```

Comment:
ReadPrimitiveTypes can be changed into setValue or setPrimitiveValue or readPrimitiveType

Reply:
Changed to `setPrimitiveValue`